test/e2e/encryption_at_rest_test.go

```diff
@@ -0,0 +1,190 @@
+	"github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/actions/kube"
+	kubecli "github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/cli/kubecli"
+	"github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/config"
+	"github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/model"
+	"github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/utils"
+)
+
+var _ = Describe("UserLogin", Label("encryption-at-rest"), func() {
+	var data model.TestDataProvider
+
+	_ = BeforeEach(func() {
+		Eventually(kubecli.GetVersionOutput()).Should(Say(K8sVersion))
+		checkUpAWSEnvironment()
+	})
+
+	_ = AfterEach(func() {
+		GinkgoWriter.Write([]byte("\n"))
+		GinkgoWriter.Write([]byte("===============================================\n"))
+		GinkgoWriter.Write([]byte("Operator namespace: " + data.Resources.Namespace + "\n"))
+		GinkgoWriter.Write([]byte("===============================================\n"))
+		if CurrentSpecReport().Failed() {
+			By("Save logs to output directory ", func() {
+				GinkgoWriter.Write([]byte("Test has been failed. Trying to save logs...\n"))
+				utils.SaveToFile(
+					fmt.Sprintf("output/%s/operatorDecribe.txt", data.Resources.Namespace),
+					[]byte(kubecli.DescribeOperatorPod(data.Resources.Namespace)),
+				)
+				utils.SaveToFile(
+					fmt.Sprintf("output/%s/operator-logs.txt", data.Resources.Namespace),
+					kubecli.GetManagerLogs(data.Resources.Namespace),
+				)
+				actions.SaveTestAppLogs(data.Resources)
+				actions.SaveK8sResources(
+					[]string{"deploy", "atlasprojects"},
+					data.Resources.Namespace,
+				)
+			})
+		}
+		By("Clean Cloud", func() {
+			DeleteAllRoles(&data)
+		})
+
+		By("Delete Resources, Project with PEService", func() {
+			actions.DeleteUserResourcesProject(&data)
+			actions.DeleteGlobalKeyIfExist(data)
+		})
+	})
+
+	DescribeTable("Namespaced operators working only with its own namespace with different configuration",
+		func(test model.TestDataProvider, encAtRest v1.EncryptionAtRest, roles []cloudaccess.Role) {
+			data = test
+			encryptionAtRestFlow(&data, encAtRest, roles)
+		},
+		Entry("Test[encryption-at-rest-aws]: Can add Encryption at Rest to AWS project", Label("encryption-at-rest-aws"),
+			model.NewTestDataProvider(
+				"encryption-at-rest-aws",
+				model.AProject{},
+				model.NewEmptyAtlasKeyType().UseDefaulFullAccess(),
+				[]string{"data/atlasdeployment_standard.yaml"},
+				[]string{},
+				[]model.DBUser{
+					*model.NewDBUser("user1").
+						WithSecretRef("dbuser-secret-u1").
+						AddBuildInAdminRole(),
+				},
+				40000,
+				[]func(*model.TestDataProvider){},
+			),
+			v1.EncryptionAtRest{
+				AwsKms: v1.AwsKms{
+					Enabled: toptr.MakePtr(true),
+					// CustomerMasterKeyID: "",
+					Region: "US_EAST_1",
+					Valid:  toptr.MakePtr(true),
+				},
+			},
+			[]cloudaccess.Role{
+				{
+					Name: utils.RandomName(awsRoleNameBase),
+					AccessRole: v1.CloudProviderAccessRole{
+						ProviderName: "AWS",
+					},
+				},
+			},
+		),
```

Comment:
It seems that it is worth adding tests for GCP and Azure.

Reply:
I don't think they are quite necessary in this initial version as operator logic is actually identical for each provider, only the object changes. We can use the same logic as for `integrations` and omit them for now.